src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Parser.cs

```diff
@@ -0,0 +1,566 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    public sealed partial class ConfigurationBindingSourceGenerator
+    {
+        private sealed class Parser
+        {
+            private readonly SourceProductionContext _context;
```

Comment:
If you're looking to cache incremental "spec" instances, then the parser should not take a dependency on `SourceProductionContext`, as this can only be supplied in the final generation callback. 

You should still be able to emit diagnostics in the parser phase by making `Diagnostic` values a part of your `SourceGenerationSpec`: `Diagnostic` implements IEquatable so incremental values can take these into account as well.